### PR TITLE
AP_Baro: Add in BMP581 Driver

### PR DESCRIPTION
**Pull request by cuav-chen2**
Add in BMP581 Driver.
Here is the data log compared with other sensors. Baro0 is the BMP581 and Baro1 is the ICP20100.
press
press2
temp
temp2

### CODE REVIEW

**Review by andyp1per — APPROVED** [review]
LGTM

**Review by tridge — APPROVED** [review]

**Comment by IamPete1** [comment]
A WebTools PR to add the new ID would be great. https://github.com/ArduPilot/WebTools/blob/ff936c1242cdf186539fcce85806aee77d80d102/Libraries/DecodeDevID.js#L107

**Comment by cuav-chen2** [comment]
Okay, I will add the PR, thanks.

**Comment by cuhome** [comment]
@andyp1per 
It has been reviewed, can you help me merge it? I will submit a PR for 7-NANO after it completes the merge.

**Comment by peterbarker** [comment]
Merged, thanks!

File: libraries/AP_Baro/AP_Baro_BMP581.cpp [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    default:
+        return false;
+    }
+
+    uint8_t status;
+    if (!_dev->read_registers(BMP581_REG_STATUS, &status, 1)) {
+        return false;
+    }
+
+    if ((status & 0b10) == 0  || (status & 0b100) == 1) {
```
Comment (peterbarker) [inline libraries/AP_Baro/AP_Baro_BMP581.cpp:117]: This line has a problem in it.  `status & 0b100` will never be `== 1`

**Pull request merged**